src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,39 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+
+        if (object.getClass().equals(Rectangle.class)) {
+            return (((getWidth() == null && rectangle.getWidth() == null)
+                    || getWidth().equals(rectangle.getWidth()))
+                    && ((getLength() == null && rectangle.getWidth() == null)
+                    || getLength().equals(rectangle.getLength()))
+                    && ((getColor() == null && rectangle.getColor() == null)
+                    || (getColor().equals(rectangle.getColor()))));
```

Comment:
what if width == null bit rectangle.width != null?
`getWidth().equals(rectangle.getWidth())` will this code work?

Reply:
No. made changes...